tests/test_recomb_collector.py

```diff
@@ -146,6 +148,19 @@ def test_simupop(make_pop, generations, popsize):
         print(x)
 
     ts.simplify()
+    tree_counts = list()
+    for _ in range(100):
+        mspts = msprime.simulate(2 * len(diploid_samples), Ne=popsize,
+                                 recombination_rate=recomb_rate,
+                                 length=max(locus_position))
+        tree_counts.append(mspts.num_trees)
+    mean = 1 / len(tree_counts) * sum(tree_counts)
+    sse = sum((t - mean) ** 2 for t in tree_counts)
+    std = math.sqrt(1 / (len(tree_counts) - 1) * sse)
+
+    assert ts.sample_size == mspts.sample_size
+    # TODO fix  expected error
+    assert abs(mean - ts.num_trees) < 3 * std
```

Comment:
this looks like the check you intended.  to verify you could replace ts by another instance of the same `msprime.simulate()` call.